### ISSUE TITLE
HTTPS_verify parameter broken since latest patch 

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2SAFE/B2HANDLE#68]
Hi,

The parameter HTTPS_verify can now be:
* bool False or True
* a path to a CA certificate file
It seems it always thinks it is a CA certificate file.
With the parameter set to True or False in the credentials file (which worked before) it does:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  modify 841/cf537936-0d42-11e6-ad5f-5254000df0ed TEST "this is a test"
Traceback (most recent call last):
  File "./epicclient2.py", line 297, in <module>
    _ARGS.func(_ARGS)
  File "./epicclient2.py", line 167, in modify
    **kvpairs)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 505, in modify_handle_value
    handlerecord_json = self.retrieve_handle_record_json(handle)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 283, in retrieve_handle_record_json
    response = self.__send_handle_get_request(handle)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 1067, in __send_handle_get_request
    resp = self.__handlesystemconnector.send_handle_get_request(handle, indices)
  File "/usr/lib/python2.6/site-packages/b2handle/handlesystemconnector.py", line 282, in send_handle_get_request
    resp = self.__session.get(url, headers=head, verify=veri, cert=self.__cert_object)
  File "/usr/lib/python2.6/site-packages/requests/sessions.py", line 476, in get
    return self.request('GET', url, **kwargs)
  File "/usr/lib/python2.6/site-packages/requests/sessions.py", line 464, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.6/site-packages/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.6/site-packages/requests/adapters.py", line 431, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: [Errno 185090050] _ssl.c:330: error:0B084002:x509 certificate routines:X509_load_cert_crl_file:system lib
```
Removing the parameter from the credentials and with a valid CA certificate for the handle server it works.

So it seems it always thinks it is a path to a CA certificate file.

**Comment by cookie33** [comment]
The credentials file sued is:
```
{
    "handle_server_url": "https://<url>:8001",
    "private_key": "/home/robertv/test/355_841_privkey.pem",
    "certificate_only": "/home/robertv/test/355_841_certificate_only.pem",
    "prefix": "841",
    "handleowner": "200:0.NA/841",
    "reverse_username": "841",
    "reverse_password": "<password>",
    "HTTPS_verify": "True"
}
```

**Comment by javaguirre** [comment]
Hi!

I've checked the usecase and tested It and It's working fine, I think It might be a different problem, we have a [unittest for this specific case](https://github.com/EUDAT-B2SAFE/B2HANDLE/blob/master/b2handle/tests/utilconfig_test.py#L16-L18) and all the tests are passing.

**Comment by cookie33** [comment]
If we change the code to:
```
def get_valid_https_verify(value):
    '''
    Get a value that can be the boolean representation of a string
    or a boolean itself and returns It as a boolean.
    If this is not the case, It returns a string.

    :value: The HTTPS_verify input value. A string can be passed as a path
            to a CA_BUNDLE certificate
    :returns: True, False or a string.
    '''
    http_verify_value = value
    bool_values = {'false': False, 'true': True}

    if isinstance(value, bool):
        http_verify_value = value
    elif isinstance(value, str) and value.lower() in bool_values.keys():
        http_verify_value = bool_values[value.lower()]

    if http_verify_value == str(True):
        http_verify_value = True
    elif http_verify_value == str(False):
        http_verify_value = False

    print http_verify_value
    print type(http_verify_value)

    return http_verify_value
```
And we return a boolean `True` or `False` or a string it works. It seems it is NEVER translated to a boolean. adding print statements to the `isinstance` shows it it is never executed. So it is never transformed to a boolean value. And that is what it has to be instead of a string.

**Comment by javaguirre** [comment]
![screen shot 2016-04-29 at 11 59 38](https://cloud.githubusercontent.com/assets/488556/14913238/e1ed26d6-0e01-11e6-995b-ef86b931c137.png)

**Comment by javaguirre** [comment]
Can you try that in Python 2.6?

**Comment by cookie33** [comment]
OK:
```
root# python
Python 2.6.6 (r266:84292, Jul 23 2015, 15:22:56) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> isinstance(True, bool)
True
```
But the parameter supplied is a string:
```
>>> value='True' 
>>> isinstance(value, bool)
```

**Comment by javaguirre** [comment]
![screen shot 2016-04-29 at 12 08 29](https://cloud.githubusercontent.com/assets/488556/14913404/1d97106a-0e03-11e6-926b-16a12c6c0b62.png)


It's clear It works, otherwise the unittests wouldn't pass. The problem is not this part. By the way, the real code of this method can be found here.

https://github.com/EUDAT-B2SAFE/B2HANDLE/blob/master/b2handle/utilconfig.py

**Comment by cookie33** [comment]
The value passed down is in `unicode` on python 2.6.

If it is changed to:
```
    if isinstance(value, bool):
        http_verify_value = value
    elif (isinstance(value, str) or isinstance(value, unicode)) and value.lower() in bool_values.keys():
        http_verify_value = bool_values[value.lower()]
```
It works in real life.

**Comment by cookie33** [comment]
I will make a pull request with the fix this afternoon.

**Comment by javaguirre** [comment]
Sorry, I don't have any machine with Python 2.6, It was deprecated eons ago and not safe to use.

I really think you SHOULD update, for security sake. :-)

**Comment by merretbuurman** [comment]
Hi, can you confirm that this issue is solved with PR #70 ? Can we close this issue?

**Comment by cookie33** [comment]
Fix merged. So close issue..
